doc/specs/#492 - Default Terminal/spec.md

```diff
@@ -0,0 +1,248 @@
+    - **NOTE:** I'm not precluding 3rd party registrations of either the delegation updated console nor the delegation terminal. It is our intention to allow either or both of these pieces to be replaced to the user's desires. The example is for brevity of our golden path and motivation for this scenario.
+1. The user is able to interact with the command-line client application as they would with the original console host.
+    - The user receives the additional benefit that short-running executions of a command-line application may not "blink in and disappear" as they do today when a user runs something like `ipconfig` from the runbox. The Terminal's default states tend to leave the tab open and say that the client has exited. This would allow a Run Dialog `ipconfig` user an improved experience over the default console host state of disappearing quickly.
+1. If any portion of the delegation fails, we will progressively degrade back to a `conhost` style Win32+GDI UX and nothing will be different from before.
+
+The settings experience includes:
+- Configuration of the delegation operations:
+    - Locations:
+        - With the registry
+            - This is what's going to be available first and will remain available. We will progress to some or all of the below after.
+            - We will need to potentially add specifications to this to both the default profile (for new installations of Windows) or to upgrade/migration profiles (for users coming from previous editions of Windows) to enable the delegation process, especially if we put a copy of Windows Terminal directly into the box.
+        - Inside Windows Terminal
+            - Inside the new Settings UI, we will likely need a page that configures the delegation keys in `HKCU\Console` or a link out to the Windows Settings panel, should we manage to get the settings configurable there.
+        - Inside the console property sheet
+            - Same as for Terminal but with `comctl` controlsover XAML +/- a link to the Windows Settings panel
+        - Inside the Settings panel for Windows (probably on the developer settings page)
+            - The ultimate location for this is likely a panel directly inside Windows. This is the hardest one to accomplish because of the timelines of the Windows product. We may not get this in an initial revision, but it should likely be our ultimate goal.
+    - Operation:
+        - Specify paths/server IDs - This is the initial revision
+        - Offer a list of registered servers or discovered manifests from the app catalog - This is the ideal scenario where we search the installed app catalog +/- the COM catalog and offer a list of apps that conform to the contract in a drop-down.
+    
+- Configuration of the legacy console state:

```

Comment:
What if launching `conhost.exe` directly _didn't_ do the delegation to the terminal? We still need a way to be able to run conhost and wt SxS, so we can debug if things regressed. 

We could still have conhost.exe do the console delegation, but then it might be harder for us to check if something regressed between windows versions, because we couldn't just plop down an old conhost on a new machine - we'd have to also remove the `ConsoleDelegation` key

Reply:
We left in the `conhost.exe cmd.exe` thing that ignores all delegations so we can have the debug corner covered.